### ISSUE TITLE
Request to allow hosting static files.

### ISSUE THREAD

**Issue by high-functioning-sociopath** [spaceuptech/space-cloud#66]
It would be really useful if space-cloud allowed the user to host static files as well.
Adding this feature will definitely help space-cloud lift up above the others.

**Comment by YourTechBud** [comment]
I think we must do that. It would be an awesome feature to have. Maybe we could have some plugins to host projects made in angular and react as well?

@high-functioning-sociopath what kind of static file hosting where you talking about? Did you mean `space-cloud` should host you website and other stuff as well? If yes, what's wrong with something like nginx and Apache web server in front of `space-cloud`?

I'm just trying to understand the use case a bit better.

**Comment by YourTechBud** [comment]
@high-functioning-sociopath ?

**Comment by high-functioning-sociopath** [comment]
Web page content files(HTML and stuff)
I was using MEAN Stack with express server to host it.
So if **space-cloud** allows these features as well, it would be amazing.
@YourTechBud

**Comment by YourTechBud** [comment]
@Jayesh26 we should definitely add this feature. Maybe space cloud hosts files by default in a public directory?

**Comment by Jayesh333** [comment]
So this will be separate module which takes a port number and a directory path to host in the config file ?

**Comment by YourTechBud** [comment]
Doesn't need to be a module as such. Let's use the same http server for this. All of our endpoints are prefixed by `/v1/api` as it is.

**Comment by Jayesh333** [comment]
At what endpoint should the static files be available? And what if user desires some other path?

**Comment by YourTechBud** [comment]
So we can have the location of the public folder in our config. Hosting would by default have a ` / ` prefix. So a file `a.jpg` at the location `/path/to/public/images/a.jpg` will be available at the url `/images/a.jpg`. Thats the normal way static hosting would work for something like node.js

Optionally we can have a routes.yaml file present at the root of the public folder which can have advanced config like what nginx does. 

Does this make sense?

**Comment by hyingreborn** [comment]
that's very useful ! Can be very easily integrated small front-end project。
It's can be config in yaml file，like this:
```
modules:
  static:
    enabled: true
    path: ./public
    gzip: true
```

**Comment by YourTechBud** [comment]
@hyingreborn you think we should add some advance routing capabilities like what nginx does?

**Comment by hyingreborn** [comment]
@YourTechBud 
I think it is enough to provide a basic support. For complex projects, it is clear that nginx is a more professional and more appropriate choice.

**Comment by YourTechBud** [comment]
Cool. So let's just start with the basic support. @hyingreborn you think you'll be able to implement this one? Or maybe @Inconnu08 you can take this up?

**Comment by Inconnu08** [comment]
Me interested to take on this. :-)

**Comment by YourTechBud** [comment]
Awesome @Inconnu08! Take it up!

**Comment by YourTechBud** [comment]
solved in #188
